### ISSUE TITLE
Submitting Homework

### ISSUE THREAD

**Issue by anders34** [uwhpsc-2016/homework1#60]
Unless this information is somewhere else, would you mind explaining again exactly what we have to do in order to submit the homework? I've finished everything else and I just don't want to mess up this last step.

**Comment by anders34** [comment]
And also, along with that, are we only pushing Exercise1.py, Exercise2.py, Exercise3.py, and report.pdf?

**Comment by cswiercz** [comment]
Use

```
$ git push
```

to push your _commits_ to the GitHub repository. The default remote should automatically be set to `https://github.com/uwhpsc-2016/homework1-anders34`. As mentioned in the email I sent to the class, if you cannot see your work at

https://github.com/uwhpsc-2016/homework1-anders34

then we cannot grade it.

**Comment by cswiercz** [comment]
We will be testing the contents of `homework/exerciseN.py` as well as your `report/report.pdf`. The tests you've written are only to your benefit.

**Comment by anders34** [comment]
So in the terminal, we first do git init, right? And then git add? And do we just add the 4 files mentioned above? then git commit and then git push?

**Comment by anders34** [comment]
also, I accidentally already added all of homework1-anders34. do I have to clear that somehow?

**Comment by cswiercz** [comment]
We talked about `git` during the first week of class and I have demonstrated in class how to submit your homework. Regardless, I will repeat the steps:
1. Obtain your private GitHub repo by clicking on the link provided via email. (You have already done this and now have ownership of https://github.com/uwhpsc-2016/homework1-anders34 )
2. Create a local clone on SMC. Based on what I see in your SMC project (which I can view) you've successfully done this step.
3. `git add` any changes you've made --- files added, edited, etc.
4. `git commit` these changes. This will create a commit with the changes you've specified.
5. This commit only lives on your SMC clone. You need to `git push` these commits to your GitHub repository.

Please review the lecture on `git`.

**Comment by cswiercz** [comment]
Additionally, you do not need to execute `git init` because the directory you cloned from GitHub is, by definition, a directory which is managed under git.

**Comment by anders34** [comment]
should the terminal I'm using to do this be inside the homework1-anders34 folder?

**Comment by cswiercz** [comment]
Yes, you need to be in the directory `homework-anders43` on SMC...

**Comment by anders34** [comment]
Alright, I think I finally figured it out. Thanks for your help!

**Comment by rhong32** [comment]
I might have made a mistake.
On my github private repository homework1-rhong32
I saw the button "Upload File". So, in the folder homework1, I uploaded my .py files for Exercise1, Exercise2, and Exercise3 and clicked on my github repository to commit this changes.
I see the files in my private repository.
I'm just not sure if this will cause any problems for grading.

Also, now, I cannot push any more changes directly from my sagemath account.

**Comment by cswiercz** [comment]
> I might have made a mistake.
>   On my github private repository homework1-rhong32
>   I saw the button "Upload File". So, in the folder homework1, I uploaded my .py files for Exercise1,     Exercise2, and Exercise3 and clicked on my github repository to commit this changes.
>   I see the files in my private repository.
>   I'm just not sure if this will cause any problems for grading.

This should be sufficient for this simple homework. but make sure to use `git push` next time.

> Also, now, I cannot push any more changes directly from my sagemath account.

I will look into this.

**Comment by cswiercz** [comment]
@rhong32 I took a look at your SMC project. Please review the demonstration of how to obtain, work on, and submit homework I gave in-class two weeks ago.
- There is no need to use `git init` once you've cloned your repo.
- Review the meanings of `git add`, `git commit` , and `git push`. Based on what I see you've added changes but not committed them.
- You cannot push un-commited changes.

You must review git before starting the next homework.
